lib/charms/observability_libs/v1/cert_handler.py

```diff
@@ -130,6 +280,17 @@ def __init__(
         self.sans_ip = list(filter(is_ip_address, sans))
         self.sans_dns = list(filterfalse(is_ip_address, sans))
 
+        if self._check_juju_supports_secrets():
+            vault_backend = _SecretVaultBackend(charm, label="cert-handler-private-vault")
```

Comment:
The `label` arg should probably be in `Vault`, not in the backend.

Reply:
Mmm I tend to disagree because it's secret-specific and Vault is backend-agnostic.
Renamed it to `secret_label` for clarity. 
Symmetrically, _RelationVaultBackend has a `relation_name` arg that is none of Vault's concern.